pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,83 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	otbridge "go.opentelemetry.io/otel/bridge/opentracing"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT   opentracing.Tracer
+	OTEL trace.TracerProvider
 }
 
-func OT(t opentracing.Tracer) JTracer {
-	return JTracer{OT: t}
+var once sync.Once
+
+func New() JTracer {
+	return JTracer{OT: initOT(), OTEL: initOTEL()}
 }
 
 func NoOp() JTracer {
-	return JTracer{OT: opentracing.NoopTracer{}}
+	return JTracer{OT: opentracing.NoopTracer{}, OTEL: trace.NewNoopTracerProvider()}
+}
+
+// InitOTEL initializes OpenTelemetry SDK.
+func initOTEL() trace.TracerProvider {
+	_, oteltp := initBoth()
+
+	return oteltp
+}
+
+// Init returns OTel-OpenTracing Bridge.
+func initOT() opentracing.Tracer {
+	otTracer, _ := initBoth()
+
+	return otTracer
+}
+
+// initBOTH initializes OpenTelemetry SDK and uses OTel-OpenTracing Bridge
+func initBoth() (opentracing.Tracer, trace.TracerProvider) {
+	opts := []otlptracehttp.Option{otlptracehttp.WithInsecure()}
+	traceExporter, err := otlptrace.New(
+		context.Background(),
+		otlptracehttp.NewClient(opts...),
+	)
+	if err != nil {
+		log.Fatal("failed to create exporter: %w", err)
+	}
+
+	// Register the trace exporter with a TracerProvider, using a batch
+	// span processor to aggregate spans before export.
+	bsp := sdktrace.NewBatchSpanProcessor(traceExporter)
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithSpanProcessor(bsp),
+		sdktrace.WithResource(resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String("otlp"),
+		)),
+	)
+
+	once.Do(func() {
+		otel.SetTextMapPropagator(
+			propagation.NewCompositeTextMapPropagator(
+				propagation.TraceContext{},
+				propagation.Baggage{},
+			))
+	})
+
+	otTracer, _ := otbridge.NewTracerPair(tracerProvider.Tracer(""))
```

Comment:
What is the 2nd value you're ignoring here? This is prod code, not sample like hotrod, we try not to ignore these.

Also, how are you going to shutdown them? Jaeger used to return Closer, OTEL might have close as a method, but your main in not calling close.

Reply:
why the 2nd value ignored?